addons/sale/models/res_partner.py

```diff
@@ -43,3 +43,11 @@ def can_edit_vat(self):
             ('state', 'in', ['sent', 'sale', 'done'])
         ], limit=1)
         return can_edit_vat and not bool(has_so)
+
+    def action_view_sale_order(self):
+        action = self.env['ir.actions.act_window']._for_xml_id('sale.act_res_partner_2_sale_order')
+        if self.is_company:
+            action['domain'] = [('partner_id.commercial_partner_id.id', '=', self.id)]
```

Comment:
don't need the `.id`
`partner_id.commercial_partner_id.id` --> `partner_id.commercial_partner_id`